src/ipelib/dynamo/module_weimer2005Ipe.f

```diff
@@ -0,0 +1,806 @@
+!-----------------------------------------------------------------------
+      module module_weimer2005Ipe
+!module w05sc
+      implicit none
+!
+! Data read from W05scEpot.dat or W05scBpot.dat:
+      integer,parameter :: csize=28, d1_pot=15, d2_pot=18
+      integer :: ab(csize), ls(csize), ms(csize)
+      real*8 :: alschfits(d2_pot,csize), schfits(d1_pot,csize)
+      real*8 :: ex_pot(2)
+      integer :: maxl_pot,maxm_pot
+!
+! Data read from SCHAtable.dat:
+      integer,parameter :: d1_scha=19, d2_scha=7, d3_scha=68
+      real*8 :: allnkm(d1_scha,d2_scha,d3_scha)
+      integer :: maxk_scha,maxm_scha
+      real*8 :: th0s(d3_scha)
+!
+! Data read from W05scBndy.dat:
+      integer,parameter :: na=6, nb=7
+      real*8 :: bndya(na),bndyb(nb),ex_bndy(2)
+!
+! Location of W05scEpot.dat, W05SCHAtable.dat, W05scBndy.dat
+!  character*7  :: file_location = 'moddta/'
+      character(len=*),parameter  :: fileLocation = './'
+!
+      real*8 :: rad2deg,deg2rad ! set by setmodel
+      real*8 :: bndyfitr        ! calculated by setboundary
+      real*8 :: esphc(csize),bsphc(csize) ! calculated by setmodel
+      real*8 :: tmat(3,3),ttmat(3,3) ! from setboundary
+      integer,parameter :: mxtablesize=200
+      real*8 :: plmtable(mxtablesize,csize)
+      real*8 :: colattable(mxtablesize)
+      real*8 :: nlms(csize)
+!
+      PRIVATE
+      PUBLIC :: setmodel2005Ipe,epotval2005Ipe,fileLocation
+      contains
+!----------------------------------------------------------------------
+!----------------------------------------------------------------------
+      subroutine setmodel2005Ipe(angle,bt,tilt,swvel,swden,file_path    &
+     &,model)
+      implicit none
+!
+! Args:
+!
+! file_path: directory in which to find data file (must have "/" at end)
+! model: must be either 'epot' or 'bpot' for electric or magnetic potential
+!
+      real*8,intent(in) :: angle,bt,tilt,swvel,swden
+      character(len=*),intent(in) :: file_path,model
+!
+! Local:
+      integer :: i,j
+      real*8 :: pi,stilt,stilt2,sw,swp,swe,c0
+      real*8 :: rang,cosa,sina,cos2a,sin2a
+      real*8 :: cfits(d1_pot,csize),a(d1_pot)
+!
+      if (trim(model) /= 'epot'.and.trim(model) /= 'bpot') then
+         write(6,"('>>> model=',a)") trim(model)
+         write(6,"('>>> setmodel: model must be either epot or bpot')")
+         stop 'setmodel' 
+      endif
+
+!   write(6,"('setmodel: angle=',f8.2,' bt=',f8.2,' tilt=',f8.2,' swvel=',&
+!     &f8.2,' swden=',f8.2)") angle,bt,tilt,swvel,swden
+!
+! Read data:
+      if (trim(model) == 'epot') then
+      call read_potential2005Ipe('global_idea_coeff_W05scEpot.dat')
+      else
+      call read_potential2005Ipe('global_idea_coeff_W05scBpot.dat')
+      endif
+      call schatable2005Ipe('global_idea_coeff_W05SCHAtable.dat')
+!
+      pi = 4.*atan(1.)
+      rad2deg = 180./pi
+      deg2rad = pi/180.
```

Comment:
These should be set as parameters and possibly shared or imported for consistency.